test/accounts/PoolManager.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import { MockHomeConnector } from "../mock/MockHomeConnector.sol";
+import "forge-std/Test.sol";
+
+contract InvariantPoolManager is Test {
+  
+  MockHomeConnector connector;
+
+  uint64[] public allPools;
+  bytes16[] public allTranches;
+  mapping(bytes16 => uint64) public trancheIdToPoolId;
+
+  constructor(MockHomeConnector connector_) public {
+    connector = connector_;
+  }
+
+  function addPool(uint64 poolId) public {
+    connector.addPool(poolId);
+    
+    allPools.push(poolId);
+  }
+
+  // uint8 so the fuzzer more often finds valid array indices
+  function addTranche(uint8 poolIndex, bytes16 trancheId, string memory tokenName, string memory tokenSymbol, uint128 price) public {
+    uint64 poolId = allPools[poolIndex];
+    vm.assume(poolId > 0);
```

Comment:
Why do we need this `assume`?

Reply:
The fuzzer will generate random uint8 `poolIndex` values, and there's no guarantee that these actually exist. This basically checks that.